stats/test/handler.test.js

```diff
@@ -767,6 +768,63 @@ describe('HTTP request handler', () => {
       ])
     })
   })
+
+  describe('miner retrieval timing stats', () => {
+    beforeEach(async () => {
+      // before the range
+      await givenRetrievalTimings(pgPools.evaluate, { day: '2024-01-09', minerId: 'f1one', timeToFirstByteP50: [1000] })
+      await givenRetrievalTimings(pgPools.evaluate, { day: '2024-01-09', minerId: 'f1two', timeToFirstByteP50: [1000] })
+      // in the range
+      await givenRetrievalTimings(pgPools.evaluate, { day: '2024-01-20', minerId: 'f1one', timeToFirstByteP50: [1000] })
+      await givenRetrievalTimings(pgPools.evaluate, { day: '2024-01-20', minerId: 'f1two', timeToFirstByteP50: [1000] })
+
+      await givenRetrievalTimings(pgPools.evaluate, { day: '2024-01-10', minerId: 'f1one', timeToFirstByteP50: [123, 345] })
+      await givenRetrievalTimings(pgPools.evaluate, { day: '2024-01-10', minerId: 'f1two', timeToFirstByteP50: [654, 789] })
+      // after the range
+      await givenRetrievalTimings(pgPools.evaluate, { day: '2024-01-21', minerId: 'f1one', timeToFirstByteP50: [1000] })
+      await givenRetrievalTimings(pgPools.evaluate, { day: '2024-01-21', minerId: 'f1two', timeToFirstByteP50: [1000] })
+    })
+
+    it('lists daily retrieval timings in given date range', async () => {
+      const res = await fetch(
+        new URL(
+          '/retrieval-timings/daily?from=2024-01-10&to=2024-01-20',
+          baseUrl
+        ), {
+          redirect: 'manual'
+        }
+      )
+      await assertResponseStatus(res, 200)
+
+      const stats = /** @type {{ day: string, success_rate: number }[]} */(
+        await res.json()
+      )
+      assert.deepStrictEqual(stats, [
+        { day: '2024-01-10', ttfb_ms: 500 },
+        { day: '2024-01-20', ttfb_ms: 1000 }
+      ])
+    })
+
+    it('lists daily retrieval timings summary for specified miner in given date range', async () => {
+      const res = await fetch(
+        new URL(
+          '/miner/f1one/retrieval-timings/summary?from=2024-01-10&to=2024-01-20',
```

Comment:
IIRC, we use `/summary` for endpoints returning aggregated data over the entire interval. This route returns data for each day within the interval. I propose to rename the route to `/miner/:id/retrieval-timings/daily`.